app/src/main/java/org/kinecosystem/kinit/view/comingSoon/EcoAppsComingSoonActivity.kt

```diff
@@ -0,0 +1,45 @@
+package org.kinecosystem.kinit.view.comingSoon
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import org.kinecosystem.kinit.KinitApplication
+import org.kinecosystem.kinit.view.SingleFragmentActivity
+import org.kinecosystem.kinit.viewmodel.EcoAppsComingSoonViewModel
+
+class EcoAppsComingSoonActivity : SingleFragmentActivity(), EcoAppsComingSoonViewModel.ComingSoonActions {
+    companion object {
+        fun getIntent(context: Context) = Intent(context, EcoAppsComingSoonActivity::class.java)
+    }
+
+    private var webfragment: EcoAppsComingSoonWebFragment? = null
+    private var model: EcoAppsComingSoonViewModel = EcoAppsComingSoonViewModel()
+
+    override fun getFragment(): Fragment {
+        webfragment = EcoAppsComingSoonWebFragment.getInstance()
+        return webfragment as EcoAppsComingSoonWebFragment
```

Comment:
why do you need to cast?

Reply:
> Smart cast to 'Type' is impossible, because 'left' is a mutable property that could have been changed by this time.

but I agree with you, this is redundant